src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,143 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int GROW_FACTOR = 2;
+    private static final int ZERO_HASH = 0;
+
+    private Entry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putNullKey(value);
+            return;
+        }
+
+        int hash = key.hashCode();
+        int index = getIndex(hash);
 
+        Entry<K, V> current = table[index];
+        while (current != null) {
+            if (current.key != null && current.key.equals(key)) {
+                current.value = value;
+                return;
+            }
+            current = current.next;
+        }
+
+        Entry<K, V> newEntry = new Entry<>(key, value);
+        newEntry.next = table[index];
+        table[index] = newEntry;
+        size++;
+
+        if ((double) size / table.length > LOAD_FACTOR) {
+            resizeTable();
+        }
```

Comment:
code duplication. Also, if resizeTable is only used inside if, move if to the method and change its name to resizeIfNeeded